client/components/data/domain-management/email/index.jsx

```diff
@@ -22,110 +22,117 @@ import {
 	getBySite,
 	isLoaded
 } from 'state/google-apps-users/selectors';
-import { shouldFetchSitePlans } from 'lib/plans';
-import { fetchSitePlans } from 'state/sites/plans/actions';
+import { getDomainsBySite, isRequestingSiteDomains } from 'state/sites/domains/selectors';
 import { getPlansBySite } from 'state/sites/plans/selectors';
+import { getProductsList } from 'state/products-list/selectors';
+import { getSelectedSite } from 'state/ui/selectors';
 
-const user = userFactory();
-
-var stores = [
-	DomainsStore,
+const stores = [
 	CartStore
 ];
 
 function getStateFromStores( props ) {
 	return {
-		domains: DomainsStore.getBySite( props.selectedSite.ID ) || {},
+		domains: props.domains,
 		cart: CartStore.get(),
 		context: props.context,
 		products: props.products,
 		selectedDomainName: props.selectedDomainName,
 		selectedSite: props.selectedSite,
 		sitePlans: props.sitePlans,
-		user: user.get(),
+		user: props.user,
 		googleAppsUsers: props.googleAppsUsers,
 		googleAppsUsersLoaded: props.googleAppsUsersLoaded
 	};
 }
 
-const EmailData = React.createClass( {
-	displayName: 'EmailData',
-
-	propTypes: {
-		component: React.PropTypes.func.isRequired,
-		context: React.PropTypes.object.isRequired,
-		productsList: React.PropTypes.object.isRequired,
-		selectedDomainName: React.PropTypes.string,
-		sitePlans: React.PropTypes.object.isRequired,
-		sites: React.PropTypes.object.isRequired,
-		googleAppsUsers: React.PropTypes.array.isRequired,
-		googleAppsUsersLoaded: React.PropTypes.bool.isRequired
-	},
-
-	mixins: [ observe( 'productsList' ) ],
+export class EmailData extends Component {
+	static propTypes = {
+		component: PropTypes.func.isRequired,
+		context: PropTypes.object.isRequired,
+		products: PropTypes.object.isRequired,
+		selectedDomainName: PropTypes.string,
+		selectedSite: PropTypes.object.isRequired,
+		sitePlans: PropTypes.object.isRequired,
+		googleAppsUsers: PropTypes.array.isRequired,
+		googleAppsUsersLoaded: PropTypes.bool.isRequired,
+	};
 
-	componentWillMount() {
-		this.loadDomainsAndSitePlans();
+	componentDidMount() {
 		this.props.fetchGoogleAppsUsers();
-	},
-
-	componentWillUpdate() {
-		this.loadDomainsAndSitePlans();
-	},
-
-	loadDomainsAndSitePlans() {
-		const selectedSite = this.props.sites.getSelectedSite();
-
-		if ( this.prevSelectedSite !== selectedSite ) {
-			fetchDomains( selectedSite.ID );
-			this.props.fetchSitePlans( this.props.sitePlans, this.props.sites.getSelectedSite() );
-
-			this.prevSelectedSite = selectedSite;
-		}
-	},
+	}
 
 	render() {
+		const {
+			selectedSite: { ID: siteId },
+		} = this.props;
```

Comment:
I was working on seeing if we could take out `selectedSite`, but it's coming in as a prop anyway right now for other uses and I don't see a reason to duplicate here by adding `siteId` as an additional prop.